dao/src/main/resources/db/migration/V75__extendVulnerabilityReferences.sql

```diff
@@ -0,0 +1,27 @@
+-- This migration script adds new columns to the vulnerability_references table that allow a better
+-- interpretation of the severity of a vulnerability. The score is now reflected as a numeric property.
+-- For existing data, this script tries to extract the numeric score from the original severity field.
+
+ALTER TABLE vulnerability_references
+    ADD COLUMN score    float NULL,
+    ADD COLUMN vector   text  NULL;
+
+-- A temporary function to handle the conversion of the severity field to a numeric score.
+CREATE FUNCTION to_number_or_null(text) RETURNS FLOAT AS $$
+DECLARE x FLOAT;
+BEGIN
+    x = $1::FLOAT;
+    RETURN x;
+EXCEPTION WHEN others THEN
+    RETURN NULL;
+END;
+$$
+STRICT
+LANGUAGE plpgsql IMMUTABLE;
+
+-- Update the score column with the numeric value of the severity field.
+UPDATE vulnerability_references
+SET score = to_number_or_null(severity);
```

Comment:
I think the `severity` column should also be set to `NULL` if it contains a numerical score. Otherwise, we would show the scores in places that expect a qualitative rating (affecting only old runs, of course).

Reply:
I am happy that I do not have to fill proper values into this column.

Reply:
Yes, while possible in theory I think it would not be worth the effort.